--- larrd-grapher.cgi	2004-09-07 15:26:51.850896037 -0700
+++ larrd-grapher-new.cgi	2004-09-08 07:40:41.324989975 -0700
@@ -915,10 +915,74 @@
             qq{GPRINT:$p:AVERAGE: \\: %5.1lf (avg)\\n},
         ]
     },
   },
 
+  mysqlslow => {
+    hourly => {
+        start_sec => "e-48h",
+        title => "$host Mysql Slow Queries Last 48 Hours",
+    },
+    daily => {
+        start_sec => "e-12d",
+        title => "$host Mysql Slow Queries Last 12 Days",
+    },
+    weekly => {
+        start_sec => "e-48d",
+        title => "$host Mysql Slow Queries Last 48 Days",
+    },
+    monthly => {
+        start_sec => "e-576D",
+        title => "$host Mysql Slow Queries Last 576 Days",
+    },
+    yaxis => "Slow Queries",
+    lines => sub {
+        my ($RRD,$service,$graph)=@_;
+        return [
+            qq{DEF:slow=$RRD:$service:AVERAGE},
+            qq{LINE2:slow#00CCCC:Slow Queries},
+            qq{COMMENT:\\n},
+            qq{GPRINT:slow:LAST: \\: %5.1lf (cur)},
+            qq{GPRINT:slow:MAX: \\: %5.1lf (max)},
+            qq{GPRINT:slow:MIN: \\: %5.1lf (min)},
+            qq{GPRINT:slow:AVERAGE: \\: %5.1lf (avg)\\n},
+        ]
+    },
+  },
+
+  mysqlthread => {
+    hourly => {
+        start_sec => "e-48h",
+        title => "$host Mysql Active Threads Last 48 Hours",
+    },
+    daily => {
+        start_sec => "e-12d",
+        title => "$host Mysql Active Threads Last 12 Days",
+    },
+    weekly => {
+        start_sec => "e-48d",
+        title => "$host Mysql Active Threads Last 48 Days",
+    },
+    monthly => {
+        start_sec => "e-576D",
+        title => "$host Mysql Active Threads Last 576 Days",
+    },
+    yaxis => "Threads",
+    lines => sub {
+        my ($RRD,$service,$graph)=@_;
+        return [
+            qq{DEF:threads=$RRD:$service:AVERAGE},
+            qq{AREA:threads#00CCCC:Active Threads},
+            qq{COMMENT:\\n},
+            qq{GPRINT:threads:LAST: \\: %5.1lf (cur)},
+            qq{GPRINT:threads:MAX: \\: %5.1lf (max)},
+            qq{GPRINT:threads:MIN: \\: %5.1lf (min)},
+            qq{GPRINT:threads:AVERAGE: \\: %5.1lf (avg)\\n},
+        ]
+    },
+  },
+
   bind => {
     hourly => {
         start_sec => "e-48h",
         title => "$host BIND Utilization Last 48 Hours",
     },
